ActivityPub.Types/Json/JsonLDContextConverter.cs

```diff
@@ -22,34 +26,57 @@ public class JsonLDContextConverter : JsonConverter<JsonLDContext>
                 return null;
 
             case JsonTokenType.String:
+            case JsonTokenType.StartObject:
             {
-                var str = reader.GetString()!;
-                return new JsonLDContext(str);
+                var context = ReadContext(ref reader, options);
+                return new JsonLDContext(new HashSet<JsonLDContextObject>
+                {
+                    context
+                } );
             }
 
-            case JsonTokenType.StartObject:
+            case JsonTokenType.StartArray:
             {
-                var terms = JsonSerializer.Deserialize<Dictionary<string, JsonLDTerm>>(ref reader, options)?.AsReadOnly();
-                if (terms == null)
-                    throw new JsonException("Failed to parse JsonLDContext terms");
+                var set = new HashSet<JsonLDContextObject>();
+                while (reader.Read() && reader.TokenType != JsonTokenType.EndArray)
+                {
+                    var context = ReadContext(ref reader, options);
+                    set.Add(context);
+                }
 
-                return new JsonLDContext(terms);
+                return new JsonLDContext(set);
             }
 
             default:
-                throw new JsonException($"Cannot deserialize {reader.TokenType} as JsonLDContext");
+                throw new JsonException($"Cannot deserialize {reader.TokenType} as @context field");
         }
     }
 
+    private static JsonLDContextObject ReadContext(ref Utf8JsonReader reader, JsonSerializerOptions options)
+    {
+        var context = JsonSerializer.Deserialize<JsonLDContextObject>(ref reader, options);
+        if (context == null)
+            throw new JsonException($"Failed to parse object-form @context field");
```

Comment:
The message should be updated - this function now handles string-form contexts as well.
```suggestion
            throw new JsonException($"Failed to parse @context field - a context was null or invalid");
```